### DISCUSSION TITLE
Problems installing Pmetrics

### DISCUSSION THREAD

**Discussion by SoniaGaG** [LAPKB/Pmetrics#205] [Q&A]
Hi!! I'm trying to install Pmetrics, and after performing the step: "type the following into the R console (lower left window of Rstudio): devtools::install_github("LAPKB/Pmetrics")", I get the following error in R:

![image](https://github.com/LAPKB/Pmetrics/assets/160844916/d58cfc90-6341-4cf6-a87a-f0fb6526677a)
[Warnings R.docx](https://github.com/LAPKB/Pmetrics/files/14372717/Warnings.R.docx)

Do you know what it means? Could you help me? Thank you very much.

**Comment by Siel** [comment]
Hi @SoniaGaG, From what I can see in the image you attached, the cause of Pmetrics not being installed is that the installation of the package `tidygraph`, one of Pmetrics dependencies failed.

I'd suggest to try installing that package independently using `install.packages("tidygraph")`. we might get more information about why the package is not getting installed.

**Reply by SoniaGaG** [reply]
Thank you very much for your response and your help. I have tried to install the tidygraph package and then the Pmetrics package, but I keep getting the following error:

![image](https://github.com/LAPKB/Pmetrics/assets/160844916/43d7c44b-a429-4b76-9d02-6166109bcaf1)

 Upon further investigation, I see that when installing the tidygraph package, the following appears:

![image](https://github.com/LAPKB/Pmetrics/assets/160844916/33503e95-0ef2-401f-9d74-e2a4969421a6)

Could it be that the tidygraph package is not being installed correctly, and that's why I'm getting an error? Could it be? I'm not sure what the problem could be.

Thank you very much again for your help, you are being very kind.

**Reply by mhovd** [reply]
Your latest error message indicates that the error is in `igraph`, please try to restart RStudio and run
```r
install.packages("igraph")
```

Then you can try to install Pmetrics as previously.

**Reply by SoniaGaG** [reply]
Thank you very much again for the information. I have tried reinstalling the tidygraph package, then the igraph package, and after Restarting R, I attempted to install Pmetrics using devtools::install_github("LAPKB/Pmetrics"), but it still doesn't work, and the following errors appear:

![image](https://github.com/LAPKB/Pmetrics/assets/160844916/57458adf-33a1-4d32-bd77-68849c04365b)

![image](https://github.com/LAPKB/Pmetrics/assets/160844916/7bb8033a-aabd-44e0-a354-f195551ff861)

Do you know why it could be?

Thank you very much again for all your help, best regards!

**Reply by Siel** [reply]
Hi @SoniaGaG It seems you have an old version of `igraph` installed in your system 1.2.6, but to get `tidygraph` installed you need at least `2.0`.
Could you please go to the updates menu and make sure `igraph` is updated:

![image](https://github.com/LAPKB/Pmetrics/assets/1023006/7a9ec1eb-6428-49fd-b5ce-3a85f1026ec6)
